pkg/exporter/sumologicexporter/otlp.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicexporter
+
+import (
+	"math"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// DecomposeHistograms decomposes any histograms present in the metric data into individual Sums and Gauges
+// This is a noop if no Histograms are present, but otherwise makes a copy of the whole structure
+// This exists because Sumo doesn't support OTLP histograms yet, and has the same semantics as the conversion to Prometheus format in prometheus_formatter.go
+func DecomposeHistograms(md pmetric.Metrics) pmetric.Metrics {
+	// short circuit and do nothing if no Histograms are present
+	foundHistogram := false
+	for i := 0; i < md.ResourceMetrics().Len(); i++ {
+		resourceMetric := md.ResourceMetrics().At(i)
+		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
+			scopeMetric := resourceMetric.ScopeMetrics().At(j)
+			for k := 0; k < scopeMetric.Metrics().Len(); k++ {
+				foundHistogram = foundHistogram || scopeMetric.Metrics().At(k).Type() == pmetric.MetricTypeHistogram
```

Comment:
Shouldn't we break out of the loops?

Reply:
Yeah, we could, not really sure which way is cleaner.

Reply:
Done.